src/tools/dtrust-tool.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * dtrust-tool.c: tool for D-Trust cards
+ *
+ * Copyright (C) 2024 mario.haustein@hrz.tu-chemnitz.de
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <string.h>
+
+// clang-format off
+#include "libopensc/opensc.h"
+#include "libopensc/cards.h"
+#include "libopensc/errors.h"
+// clang-format on
```

Comment:
if some of the includes need to come before others, its easier to separate them by a newline than turning off the clang format altogether.
If it is the `opensc.h`, something like this should do:
```suggestion
#include "libopensc/opensc.h"

#include "libopensc/cards.h"
#include "libopensc/errors.h"
```

Reply:
fixed ... and learned that empty lines matter in this case.